contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -3608,6 +3608,25 @@ bbf_ProcessUtility(PlannedStmt *pstmt,
 				GrantStmt *grant = (GrantStmt *) parsetree;
 				char	   *dbname = get_cur_db_name();
 				const char *current_user = GetUserNameFromId(GetUserId(), false);
+				int16 privilege_maskInt = 0;
+				ListCell   *lc1;
+				/* create integer bitmask for permissions*/
+						foreach(lc1, grant->privileges)
+						{
+							AccessPriv *ap = (AccessPriv *) lfirst(lc1);
+							if (!strcmp(ap->priv_name,"execute"))
+								privilege_maskInt |= 32;
+							if (!strcmp(ap->priv_name,"select"))
```

Comment:
if execute privilege is in the privileges list the will set a bit in the final privilege_maskInt variable at it's 0th index from left.